src/TextForEvent.js

```diff
@@ -142,9 +141,19 @@ function textForCallAnswerEvent(event) {
 }
 
 function textForCallHangupEvent(event) {
-    var senderName = event.sender ? event.sender.name : _t('Someone');
-    var supported = MatrixClientPeg.get().supportsVoip() ? "" : _t('(not supported by this browser)');
-    return _t('%(senderName)s ended the call.', {senderName: senderName}) + ' ' + supported;
+    const senderName = event.sender ? event.sender.name : _t('Someone');
+    const eventContent = event.getContent();
+    let reason = "";
+    if(!MatrixClientPeg.get().supportsVoip()) {
+      reason = _t('(not supported by this browser)');
+    } else if(eventContent.reason) {
+      if (eventContent.reason === "ice_failed") {
+        reason = _t('(call failed to connect)');
```

Comment:
"ice_failed" is a very specific error message - I'd suggest "could not connect media" rather than "call failed to connect", which sounds more like a signalling problem to me.